leveldb.c

```diff
@@ -881,7 +889,8 @@ PHP_METHOD(LevelDB, getApproximateSizes)
 	zend_hash_internal_pointer_reset_ex(Z_ARRVAL_P(limit), &pos_limit);
 
 	while ((start_val = zend_hash_get_current_data_ex(Z_ARRVAL_P(start), &pos_start)) != NULL &&
-		(limit_val = zend_hash_get_current_data_ex(Z_ARRVAL_P(limit), &pos_limit)) != NULL) {
+		(limit_val = zend_hash_get_current_data_ex(Z_ARRVAL_P(limit), &pos_limit)) != NULL &&
+		i < num_ranges) {
```

Comment:
This change seems redundant, num_ranges is inferred from the size of the input array anyway